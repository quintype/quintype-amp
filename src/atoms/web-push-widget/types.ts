import { ReactElement } from "react";

export interface WebPushWidgetTypes {
  visibility: "unsubscribed" | "subscribed" | "blocked";
  children: JSX.Element[] | JSX.Element | ReactElement | HTMLElement;
}
