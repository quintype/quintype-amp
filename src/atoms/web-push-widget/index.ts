import { WebPushWidget } from "./web-push-widget";
export { WebPushWidget };
