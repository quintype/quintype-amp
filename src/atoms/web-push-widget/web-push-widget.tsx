import React from "react";
import { WebPushWidgetTypes } from "./types";

export const WebPushWidget = (props: WebPushWidgetTypes) => {
  return <amp-web-push-widget layout="fixed" {...props} />;
};
