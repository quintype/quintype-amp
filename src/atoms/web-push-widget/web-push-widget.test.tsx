import React from "react";
import { WebPushWidget } from "./web-push-widget";
import { shallow } from "enzyme";

const wrapper = shallow(
  <WebPushWidget visibility="unsubscribed" width="350px" height="60px">
    <button>Click me</button>
    <div>I am WebPushWidget's child</div>
  </WebPushWidget>
);

describe("WebPushWidget", () => {
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should pass all props down to amp-web-push-widget", () => {
    expect(wrapper.find("amp-web-push-widget").prop("visibility")).toBe("unsubscribed");
    expect(wrapper.find("button").text()).toBe("Click me");
    expect(wrapper.find("div").text()).toBe("I am WebPushWidget's child");
  });
});
