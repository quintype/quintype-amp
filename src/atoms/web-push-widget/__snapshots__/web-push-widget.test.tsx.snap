// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebPushWidget should render 1`] = `
<amp-web-push-widget
  visibility="unsubscribed"
>
  <button>
    Click me
  </button>
  <div>
    I am WebPushWidget's child
  </div>
</amp-web-push-widget>
`;
